modules/ROOT/pages/database-internals/checkpointing.adoc

```diff
@@ -0,0 +1,155 @@
+Checkpointing involves the inclusion of a special _Checkpointing_ entry in the transaction log, marking the last transaction at which checkpointing occurred.
+This entry serves the purpose of identifying transaction logs that are no longer necessary, as all the transactions they contain have been securely stored in the storage files.
+
+The process of eliminating transaction logs that are no longer required for recovery is known as _pruning_.
+Pruning is reliant on checkpointing.
+Checkpointing determines which logs can be pruned and determines the occurrence of pruning, as the absence of a checkpoint implies that the set of transaction log files available for pruning cannot have changed.
+Consequently, pruning is triggered whenever checkpointing occurs.
+
+[[checkpointing-policy]]
+== Configure the checkpointing policy
+
+The checkpointing policy, which is the driving event for log pruning is configured by xref:configuration/configuration-settings.adoc#config_db.checkpoint[`db.checkpoint`].
+Depending on your needs, the checkpoint can run on a periodic basis, which is the default, when a certain amount of data has been written to the transaction log, or continuously.
+
+.Available checkpointing policies
+[options="header", cols="1m,3a"]
+|===
+|Policy
+|Description
+
+|PERIODIC
+|label:default[Default]
+This policy checks every 10 minutes whether there are changes pending flushing and if so, it performs a checkpoint and subsequently triggers a log prune.
+The periodic policy is specified by the `<<config_db.checkpoint.interval.tx,db.checkpoint.interval.tx>>` and `<<config_db.checkpoint.interval.time,db.checkpoint.interval.time>>` settings and the checkpointing is triggered when either of them is reached.
+See <<checkpoint-interval, Configure the checkpoint interval>> for more details.
+
+|VOLUME
+|This policy runs a checkpoint when the size of the transaction logs reaches the value specified by the `<<config_db.checkpoint.interval.volume,db.checkpoint.interval.volume>>` setting.
+By default, it is set to `250.00MiB`.
+
+|CONTINUOUS
+|label:enterprise[Enterprise Edition]
+This policy ignores `<<config_db.checkpoint.interval.tx,db.checkpoint.interval.tx>>` and `<<config_db.checkpoint.interval.time,db.checkpoint.interval.time>>` settings and runs the checkpoint process all the time.
+The log pruning is triggered immediately after the checkpointing completes, just like in the periodic policy.
+
+|VOLUMETRIC
+|label:enterprise[Enterprise Edition]
+This policy checks every 10 seconds if there is enough volume of logs available for pruning and, if so, it triggers a checkpoint and subsequently, it prunes the logs.
+By default, the volume is set to 250MiB, but it can be configured using the settings xref:configuration/configuration-settings.adoc#config_db.tx_log.rotation.retention_policy[`db.tx_log.rotation.retention_policy`] and xref:configuration/configuration-settings.adoc#config_db.tx_log.rotation.size[`db.tx_log.rotation.size`].
```

Comment:
@mnssn, Is this sentence true?

Reply:
It is true that `db.tx_log.rotation.retention_policy` is used to configure volumetric.

For `db.tx_log.rotation.size` the interplay is more subtle. This setting specifies how much space a single file may roughly occupy. This influences how much space can be reclaimed by **all** strategies by virtue of the following.

To reclaim a given file, the newest checkpoint for the transaction log must exist in another file. So if you have a huge transaction log, then it is likely that your latest checkpoint is in the same file, making it impossible to reclaim said file.